keps/sig-apps/20190226-maxunavailable-for-statefulsets.md

```diff
@@ -48,46 +48,47 @@ The purpose of this enhancement is to implement maxUnavailable for StatefulSet d
 3. My Stateful clustered application, has followers and leaders, with followers being many more than 1. My application can tolerate many followers going 
 down at the same time. I want to be able to do faster rollouts by bringing down 2 or more followers at the same time. This is only possible if StatefulSet
 supports maxUnavailable in Rolling Updates.
-4. Sometimes i just want easier tracking of revisions of a rolling update. Deployment does it through ReplicaSets and has its own nuances. Understanding 
-that requires diving into the complicacy of hashing and how replicasets are named. Over and above that, there are some issues with hash collisions which 
-further complicate the situation(I know they were solved). StatefulSet introduced ControllerRevisions in 1.7 which I believe are easier to think and reason 
-about. They are used by DaemonSet and StatefulSet for tracking revisions. It would be so much nicer if all the use cases of Deployments can be met and we 
-could track the revisions by ControllerRevisions.
+4. Sometimes I just want easier tracking of revisions of a rolling update. Deployment does it through ReplicaSets and has its own nuances. Understanding 
+that requires diving into the complicacy of hashing and how ReplicaSets are named. Over and above that, there are some issues with hash collisions which 
+further complicate the situation(I know they were solved). StatefulSet introduced ControllerRevisions in 1.7 which are much easier to think and reason 
+about. They are used by DaemonSet and StatefulSet for tracking revisions. It would be so much nicer if all the use cases of Deployments can be met in 
+StatefulSet's and additionally we could track the revisions by ControllerRevisions.
 
-With this feature in place, when using StatefulSet with maxUnavailable >1, the user understands that this would not cause issues with their Stateful 
-Applications which have per pod state and identity while still providing all of the above written advantages.
+With this feature in place, when using StatefulSet with maxUnavailable >1, the user is making a conscious choice that more than one pod going down at the 
+same time during rolling update, would not cause issues with their Stateful Application which have per pod state and identity. Other Stateful Applications
+which cannot tolerate more than one pod going down, will resort to the current behavior of one pod at a time Rolling Updates.
 
 ### Goals
 StatefulSet RollingUpdate strategy will contain an additional parameter called `maxUnavailable` to control how many Pods will be brought down at a time,
 during Rolling Update.
 
 ### Non-Goals
-maxUnavailable is only implemeted to affect the Rolling Update of StatefulSet. Considering maxUnavailable for Pod Management Policy of Parallel is beyond 
+maxUnavailable is only implemented to affect the Rolling Update of StatefulSet. Considering maxUnavailable for Pod Management Policy of Parallel is beyond 
```

Comment:
Why?

Reply:
@smarterclayton because  podManagementPolicy is only applicable during StatefulSet creation, deletion and scaling. The main use case is faster rolling updates. We could make it applicable to create/delete/scale, but that is secondary to the goal.  I thought we could just limit the scope of the initial implementation to make it less confusing, but if you think we should apply this to both, that is fine as well.

Reply:
So during upgrade, if 3 of 5 nodes are evacuated, PMP is ignored?  Since this is mostly established behavior of replicasets and deployments, you should mention *why* it's out of scope.  I am more concerned with creating something that differs between RS/Deployment and StatefulSet right now that I am about implementation ordering, so I just want the KEP to reflect that it is a) consistent with user expectations and b) justifies statements about timing with reasons